src/PrizeVault.sol

```diff
@@ -538,25 +583,21 @@ contract PrizeVault is TwabERC20, Claimable, IERC4626, ILiquidationSource, Ownab
             _yieldFee = (_amountOut * FEE_PRECISION) / (FEE_PRECISION - _yieldFeePercentage) - _amountOut;
         }
 
-        // Ensure total liquidation does not exceed the available yield balance:
-        uint256 _totalLiquidation = _amountOut + _yieldFee;
-        if (_totalLiquidation > _availableYield) {
-            revert LiquidationExceedsAvailable(_totalLiquidation, _availableYield);
+        // Ensure total liquidation amount does not exceed the available yield balance:
+        if (_amountOut + _yieldFee > _availableYield) {
+            revert LiquidationExceedsAvailable(_amountOut + _yieldFee, _availableYield);
         }
 
+        // Increase yield fee balance:
+        if (_yieldFee > 0) {
+            yieldFeeBalance += _yieldFee;
+        }
+
+        // Mint or withdraw amountOut to `_receiver`:
         if (_tokenOut == address(_asset)) {
-            // Withdraw assets:
-            _withdraw(address(this), _totalLiquidation);
-            _asset.transfer(_receiver, _amountOut);
-            if (_yieldFee > 0) {
-                _asset.transfer(_yieldFeeRecipient, _yieldFee);
-            }
+            _withdraw(_receiver, _amountOut);            
         } else if (_tokenOut == address(this)) {
-            // Mint shares:
             _mint(_receiver, _amountOut);
-            if (_yieldFee > 0) {
-                _mint(_yieldFeeRecipient, _yieldFee);
-            }
         } else {
             revert LiquidationTokenOutNotSupported(_tokenOut);
         }
```

Comment:
I think we should emit an event for transferTokensOut; feels like we're missing one in retrospect

Reply:
Added `TransferYieldOut` event here: https://github.com/GenerationSoftware/pt-v5-vault/pull/75/commits/a2c73dada1ad2cf5276b1966722217479f83899c#diff-3f32973f6e271309aabc2174e0559d7ed0b537ec3e75f6c34022b3db7d083190R626